beacon-chain/casper/incentives.go

```diff
@@ -36,8 +34,8 @@ func CalculateRewards(
 				if voterIndex == validatorIndex {
 					voted = true
 					balance := validators[validatorIndex].GetBalance()
-					newbalance := uint64(balance + (balance/rewardQuotient)*depositFactor)
-					validators[validatorIndex].Balance = newbalance
+					newbalance := int64(balance) + int64(balance/rewardQuotient)*(2*int64(totalParticipatedDeposit)-int64(totalDeposit))/int64(totalDeposit)
```

Comment:
Why conversion to int64 rather than uint64?

Reply:
so according to the spec ```balance/rewardQuotient*(2*totalParticipatedDeposit-totalDeposit/totalDeposit``` can also be negative so casting to `int64` is required over here